apps/scan/backend/src/scanners/custom/app_scan_jam.test.ts

```diff
@@ -0,0 +1,171 @@
+import { AdjudicationReason } from '@votingworks/types';
+import { ok } from '@votingworks/basics';
+import { mocks } from '@votingworks/custom-scanner';
+import {
+  configureApp,
+  expectStatus,
+  mockInterpretation,
+  waitForStatus,
+} from '../../../test/helpers/shared_helpers';
+import { SheetInterpretation } from '../../types';
+import { ballotImages, withApp } from '../../../test/helpers/custom_helpers';
+
+jest.setTimeout(20_000);
+jest.mock('@votingworks/ballot-encoder', () => {
+  return {
+    ...jest.requireActual('@votingworks/ballot-encoder'),
+    // to allow changing election definitions without changing the image fixtures
+    // TODO: generate image fixtures from election definitions more easily
+    // this election hash is for the famous names image fixtures
+    sliceElectionHash: () => 'da81438d51136692b43c',
+  };
+});
+
+const needsReviewInterpretation: SheetInterpretation = {
+  type: 'NeedsReviewSheet',
+  reasons: [{ type: AdjudicationReason.BlankBallot }],
+};
+
+test('jam on scan', async () => {
+  await withApp(
+    {
+      delays: {
+        DELAY_RECONNECT_ON_UNEXPECTED_ERROR: 500,
+      },
+    },
+    async ({ apiClient, mockScanner, mockUsb, mockAuth }) => {
+      await configureApp(apiClient, mockUsb, { mockAuth });
+
+      mockScanner.getStatus.mockResolvedValue(ok(mocks.MOCK_READY_TO_SCAN));
+      await waitForStatus(apiClient, { state: 'ready_to_scan' });
+
+      await apiClient.scanBallot();
+      mockScanner.getStatus.mockResolvedValue(ok(mocks.MOCK_INTERNAL_JAM));
+      await waitForStatus(apiClient, {
+        state: 'recovering_from_error',
+        error: 'client_error',
+      });
+
+      mockScanner.getStatus.mockResolvedValue(ok(mocks.MOCK_NO_PAPER));
+      await waitForStatus(apiClient, { state: 'no_paper' });
+    }
+  );
+});
+
+test('jam on accept', async () => {
+  await withApp(
+    {
+      delays: {
+        DELAY_ACCEPTING_TIMEOUT: 500,
+      },
+    },
+    async ({ apiClient, mockScanner, interpreter, mockUsb, mockAuth }) => {
+      await configureApp(apiClient, mockUsb, { mockAuth });
+
+      mockScanner.getStatus.mockResolvedValue(ok(mocks.MOCK_READY_TO_SCAN));
+      await waitForStatus(apiClient, { state: 'ready_to_scan' });
+
+      const interpretation: SheetInterpretation = {
+        type: 'ValidSheet',
+      };
+      mockInterpretation(interpreter, interpretation);
+
+      mockScanner.scan.mockResolvedValue(ok(await ballotImages.completeBmd()));
+      await apiClient.scanBallot();
+      await waitForStatus(apiClient, { state: 'scanning' });
+      mockScanner.getStatus.mockResolvedValue(ok(mocks.MOCK_READY_TO_EJECT));
+      await waitForStatus(apiClient, {
+        state: 'ready_to_accept',
+        interpretation,
+      });
+
+      await apiClient.acceptBallot();
+      await waitForStatus(apiClient, { state: 'accepting', interpretation });
+      // mockScanner.getStatus.mockResolvedValue(ok(mocks.MOCK_INTERNAL_JAM));
```

Comment:
Should this be uncommented?
Actually a bit confused why the rest of the test passes without this - or am I misreading? Do we automatically move into `'rejecting'` state after a certain amount of time?

Reply:
I think it was right without that line, so I'm removing it. We do transition to `rejecting` after a while if the status doesn't transition to `NO_PAPER`, and since we stay at `READY_TO_EJECT` it'll eventually transition to `rejecting`.